src/Mvc/Mvc.Razor/src/RazorPagePropertyActivator.cs

```diff
@@ -108,9 +108,9 @@ internal ViewDataDictionary CreateViewDataDictionary(ViewContext context)
             {
                 valueAccessor = valueAccessors.JsonHelperAccessor;
             }
-            else if (property.PropertyType == typeof(DiagnosticSource))
+            else if (property.PropertyType == typeof(DiagnosticListener))
```

Comment:
This is breaking, it should support both.

Reply:
Returned